test/functional/test_framework/util.py

```diff
@@ -592,3 +592,10 @@ def find_vout_for_address(node, txid, addr):
         if addr == tx["vout"][i]["scriptPubKey"]["address"]:
             return i
     raise RuntimeError("Vout not found for address: txid=%s, addr=%s" % (txid, addr))
+
+
+def sync_txindex(test_framework, node):
```

Comment:
nit: I would have put this into `test_framework/test_framework.py` so it can live among its peers `sync_blocks` et al and we would save the import from util. But fine to keep as is.

Reply:
Right, I'm not passionate about the naming here. `sync_blocks` is syncing between two blockchains, `sync_txindex` is syncing between a blockchain and an index, so it's not wrong IMO. But `wait_for_txindex()` sounds fine, too. So up to you what you want to choose and if you want to change this at all.